src/cargo/sources/registry/index.rs

```diff
@@ -344,11 +438,297 @@ impl<'cfg> RegistryIndex<'cfg> {
     }
 
     pub fn is_yanked(&mut self, pkg: PackageId, load: &mut dyn RegistryData) -> CargoResult<bool> {
-        let summaries = self.summaries(pkg.name().as_str(), load)?;
-        let found = summaries
-            .iter()
-            .filter(|&(summary, _yanked)| summary.version() == pkg.version())
-            .any(|(_summary, yanked)| *yanked);
+        let req = VersionReq::exact(pkg.version());
+        let found = self
+            .summaries(pkg.name(), &req, load)?
+            .any(|summary| summary.yanked);
         Ok(found)
     }
 }
+
+impl Summaries {
+    /// Parse out a `Summaries` instances from on-disk state.
+    ///
+    /// This will attempt to prefer parsing a previous cache file that already
+    /// exists from a previous invocation of Cargo (aka you're typing `cargo
+    /// build` again after typing it previously). If parsing fails or the cache
+    /// isn't found, then we take a slower path which loads the full descriptor
+    /// for `relative` from the underlying index (aka typically libgit2 with
+    /// crates.io) and then parse everything in there.
+    ///
+    /// * `last_index_update` - this is a file modification time where if any cache
```

Comment:
This looks great. We could be more aggressive and ask git when the last commit to use this file was, but I doubt the additional cashing will pay for the git calls.

I think some of the comments did not get updated, specifically here and in the file format description below.

I will have new timings in a fuwe.